src/changes.rs

```diff
@@ -148,6 +150,26 @@ impl<'a> ChangeSet<'a> {
                       -> Result<Option<String>, ::std::io::Error> {
         let text = &self.file_map[filename];
 
+        fn style<T>(mut writer: T,
```

Comment:
This function should also probably take writer by mutable reference, rather than value (not 100% sure if it makes much difference)